src/Splunk.OpenTelemetry.AutoInstrumentation/ResourceConfigurator.cs

```diff
@@ -57,5 +63,50 @@ public static void Configure(ResourceBuilder resourceBuilder, PluginSettings set
         };
 
         resourceBuilder.AddAttributes(attributes);
+
+#if NET6_0_OR_GREATER
+        var resource = resourceBuilder.Build();
+
+        if (resource.Attributes.All(kvp => kvp.Key != ServiceNameAttribute))
+        {
+            var localResourceBuilder = ResourceBuilder.CreateEmpty();
+            localResourceBuilder.AddAttributes(resource.Attributes);
+            // service.name was not configured yet use the fallback.
+            localResourceBuilder.AddAttributes(new KeyValuePair<string, object>[] { new(ServiceNameAttribute, ServiceNameConfigurator.GetFallbackServiceName()) });
+            SampleExporter.SetResources(localResourceBuilder.Build());
+        }
+        else
+        {
+            SampleExporter.SetResources(resource);
+        }
+#endif
```

Comment:
I have a rather strong opinion that this https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/a13345d6683768004625e860691befecfb3998eb/src/OpenTelemetry.AutoInstrumentation/Configurations/ResourceConfigurator.cs#L50-L54 should be invoked before `resourceBuilder.InvokePlugins(pluginManager);`. It would simplify this code base as well as upstream.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3203

This is what I have in mind: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3203

Reply:
PR merged. Changes in d8fc6e4e2427ccb71e2a33830d0d983f7c8cbb98